### ISSUE TITLE
When using nix-shell in a shebang with an implicit nix file lookup, it gets unhappy

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nix#1344]
This worked in 1.11:

shell.nix:
```nix
(import <nixpkgs> {}).hello
```

test.sh:
```bash
#!/usr/bin/env nix-shell
#! nix-shell -i bash
echo hi
```

What happens:

```
$ ./test.sh 
error: not an absolute path: ‘./shell.nix’
```

Same thing happens if I rename shell.nix to default.nix.

**Comment by knedlsepp** [comment]
Is there some workaround for this?

**Comment by coretemp** [comment]
For me, there is a difference on `master` between running `./test.sh` and `bash test.sh`.

The latter works. The former outputs:
`error: getting status of /home/coretemp/default.nix : No such file or directory ` 

What is it supposed to do? 

Extra information:
```
echo $SHELL
/run/current-system/sw/bin/bash
```

**Comment by coretemp** [comment]
@dtzWill Can you provide me with any treeish of nix that works including a human readable version like if this is Nix 2.0 or Nix 1.something? I am using the 17.09 release channel and this doesn't work. That has nix-1.11.6.

**Comment by dtzWill** [comment]
Don't have it handy, but pretty sure it was Nix 2 I was referring to above. Do you need something more specific?

**Comment by coretemp** [comment]
@dtzWill That's good enough. Thanks.

**Comment by roberth** [comment]
This appears to have been fixed some time ago.

Additionally, a similar issue with relative paths in e.g. `--arg` and `--expr` has been [fixed in Nix 2.24](https://nix.dev/manual/nix/2.24/release-notes/rl-2.24)
.
